.github/workflows/ci.yml

```
@@ -43,8 +44,11 @@ jobs:
       - uses: actions/checkout@v4
       - name: Cache the store
         uses: ./
+        with:
+          _internal-strict-mode: true
 
   run-x86_64-linux-untrusted:
+    needs: build
     name: Run x86_64 Linux, Untrusted
```

nit: let's keep this in line with the others in the matrix:
```suggestion
    name: Run x86_64-linux, Untrusted
```